backend/plugin/src/main/kotlin/com/ritense/plugin/service/PluginService.kt

```
@@ -280,23 +283,33 @@ class PluginService(
+        val usages = pluginConfigurationUsageResolver.findUsagesForConfiguration(pluginConfigurationId)
+        if (usages.isNotEmpty()) {
+            throw PluginConfigurationInUseException(pluginConfigurationId.id, usages)
+        }
 
-                val event = PluginConfigurationDeletedEvent(it)
-                applicationEventPublisher.publishEvent(event)
-            }
-            ?: logger.warn { "Plugin configuration with Id: [$pluginConfigurationId] was not found." }
+        try {
+            configuration.runAllPluginEvents(EventType.DELETE)
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)